indexer/services/ender/__tests__/scripts/scripts.test.ts

```diff
@@ -236,6 +315,174 @@ describe('SQL Function Tests', () => {
+      expectedResult: -1,
+    },
+    {
+      event: { blockEvent: '3' },
+      expectedError: 'Received V4 event with invalid block event type: 3',
+    },
+    {
+      event: {},
+      expectedError: 'Either transactionIndex or blockEvent must be defined in IndexerTendermintEvent',
+    },
+  ])('dydx_tendermint_event_to_transaction_index should return the expected result', async (
+    { event, expectedResult, expectedError },
+  ) => {
+    try {
+      const result = await getSingleRawQueryResultRow(
+        `SELECT dydx_tendermint_event_to_transaction_index('${JSON.stringify(event)}') AS result`);
+      if (expectedError) {
+        throw new Error('Expected an error but got a result.');
+      }
+      expect(result).toEqual(expectedResult);
+    } catch (error) {
+      if (!expectedError) {
+        throw error;
+      }
+      expect(error.message).toContain(expectedError);
+    }
+  });
+
+  it.each([
+    {
+      blockHeight: '123456',
+      transactionIndex: 123,
+    },
+  ])('dydx_uuid_from_transaction_parts (%s)', async (transactionParts) => {
+    const result = await getSingleRawQueryResultRow(
+      `SELECT dydx_uuid_from_transaction_parts('${transactionParts.blockHeight}', '${transactionParts.transactionIndex}') AS result`);
+    expect(result).toEqual(
+      TransactionTable.uuid(transactionParts.blockHeight, transactionParts.transactionIndex),
+    );
+  });
+
+  it('dydx_create_transaction.sql should insert a transaction and return correct jsonb', async () => {
+    const transactionHash: string = 'txnhash';
+    const blockHeight: string = '1';
+    const transactionIndex: number = 123;
+
+    const returnedJsonb = await getSingleRawQueryResultRow(`SELECT dydx_create_transaction('${transactionHash}', '${blockHeight}', ${transactionIndex}) AS result`);
+
+    const transactions: TransactionFromDatabase[] = await TransactionTable.findAll(
+      {},
+      [],
+      { readReplica: true },
+    );
+
+    expect(transactions.length).toEqual(1);
+    const transaction = transactions[0];
+    expect(transaction).toEqual(expect.objectContaining({
+      transactionHash,
+      blockHeight,
+      transactionIndex,
+    }));
+    expect(returnedJsonb).toEqual(expect.objectContaining({
+      transactionHash,
+      blockHeight: Number(blockHeight),
+      transactionIndex,
+    }));
+  });
+
+  it('dydx_create_tendermint_event.sql should insert a tendermint event and return correct jsonb', async () => {
+    await BlockTable.create(testConstants.defaultBlock);
+    const transactionIndex: number = 0;
+    const eventIndex: number = 0;
+
+    const indexerTendermintEvent: IndexerTendermintEvent = createIndexerTendermintEvent(
+      DydxIndexerSubtypes.ASSET,
+      binaryToBase64String(
+        AssetCreateEventV1.encode(defaultAssetCreateEvent).finish(),
+      ),
+      transactionIndex,
+      eventIndex,
+    );
+
+    const result = await getSingleRawQueryResultRow(
```

Comment:
Nit: type

Reply:
returned object is untyped. There's no precedent for casting return object to a specific type for this function.